cylc/flow/platforms.py

```diff
@@ -63,10 +70,21 @@ def get_platform(task_conf=None, task_id='unknown task', warn_only=False):
         output = platform_from_name(task_conf)
 
     elif 'platform' in task_conf and task_conf['platform']:
+        if PLATFORM_REC_COMMAND.match(task_conf['platform']) and warn_only:
+            # In warning mode this function might have been passed an
+            # un-expanded platform string - warn that they won't deal with
+            # with this until job submit.
+            return None
+        if HOST_REC_COMMAND.match(task_conf['platform']) and warn_only:
```

Comment:
This could be annoying if a user has systematically used backticks, but the tasks will fail later anyway, so we  might as well be annoying at validation.